### PR TITLE
nixos/nginx: add defaultMimeTypes option

### PR DESCRIPTION
**Pull request by Izorkin**
###### Description of changes
Add defaultMimeTypes to enable load custom mime-types.
The mime types in mailcap package have not been updated for a long time.
This PR gives the ability to load own list with mime types.

cc @dasJ @ajs124 @SuperSandro2000 @RaitoBezarius
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fpletz — APPROVED** [review]

**Comment by fpletz** [comment]
Duplicate: #208785

**Comment by RaitoBezarius** [comment]
Thank you for the contribution, I see that @SuperSandro2000 made a similar one, can you work together towards this semantic change?

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 1]
```diff
@@ -529,6 +527,18 @@ in
         '';
       };
 
+      defaultMimeTypes = mkOption {
+        type = types.str;
+        default = "${pkgs.mailcap}/etc/nginx/mime.types";
+        defaultText = literalExpression "$''{pkgs.mailcap}/etc/nginx/mime.types";
+        example = literalExpression "$''{pkgs.nginx}/conf/mime.types";
+        description = lib.mdDoc ''
+          The mime type definitions included with nginx are very incomplete, so
```
Comment (RaitoBezarius) [inline nixos/modules/services/web-servers/nginx/default.nix:536]: The description seems strange, here's my proposal:
```suggestion
          Default MIME types for NGINX, as MIME types definitions from NGINX are very incomplete,
          we use by default the ones bundled in the mailcap package, used by most of the other Linux distributions.
```

Comment (Izorkin) [reply nixos/modules/services/web-servers/nginx/default.nix]: Fixed.

**Comment by RaitoBezarius** [comment]
Ping @SuperSandro2000 @Izorkin ; were you able to sort out your changes together?

**Comment by Izorkin** [comment]
@SuperSandro2000 close this PR? In your variant, should probably move the comment from `commonHttpConfig` to `mimeTypesFile.description`

**Review by fpletz — APPROVED** [review]

File: nixos/modules/services/web-servers/nginx/default.nix [hunk 2]
```diff
@@ -529,6 +527,18 @@ in
         '';
       };
 
+      defaultMimeTypes = mkOption {
+        type = types.str;
```
[the remark this answers is not in the document]
Comment (Izorkin) [reply nixos/modules/services/web-servers/nginx/default.nix]: Fixed.

**Comment by fpletz** [comment]
Thanks for the quick fix. Let's merge. :smiley:

**Pull request merged**